src/server/GameServer.ts

```diff
@@ -290,12 +291,16 @@ export class GameServer {
     this.gameStartInfo = GameStartInfoSchema.parse({
       gameID: this.id,
       config: this.gameConfig,
-      players: this.activeClients.map((c) => ({
-        playerID: c.playerID,
-        username: c.username,
-        clientID: c.clientID,
-        flag: c.flag,
-      })),
+      players: this.activeClients.map(
+        (c) =>
+          ({
+            playerID: c.playerID,
+            username: c.username,
+            clientID: c.clientID,
+            pattern: c.pattern,
+            flag: c.flag,
+          }) satisfies Player,
```

Comment:
Sorry for the confusion. This one is redundant because GameStartInfo.players is already `Player[]`.

```suggestion
          }),
```